gr-qtgui/lib/sink_f_impl.cc

```diff
@@ -135,8 +135,18 @@ QWidget* sink_f_impl::qwidget() { return d_main_gui.qwidget(); }
 
 void sink_f_impl::set_fft_size(const int fftsize)
 {
-    d_fftsize = fftsize;
-    d_main_gui.setFFTSize(fftsize);
+    if ((fftsize >= d_main_gui.MIN_FFT_SIZE) && (fftsize <= d_main_gui.MAX_FFT_SIZE)) {
```

Comment:
My expectation would have been to coerce to the closest available value (e.g., if you request FFT size 10000 you still get 32kib). But honestly, I'm not sure if that's better than this, so no complaint here, just an observation.

Reply:
At the moment, the power-of-two check is only implemented in freq_sink_(c/f). But you can only enter a non power-of-two value, if you edit the python code. In this case 10000 would be replaced by 8192. Another possible value could be 16384.

If you use grc, you can't generate a flowgraph wit fftsize of 10000.
I'd like to handle all the fftsize checking and possible replacement in a single class, that could be used by the different sinks. This would make maintaining easier. But this will take some time. So the assertions in the yml files will help to avoid some but not all possible errors.